tests/Integration/Behaviour/Features/Context/Domain/ProductFeatureContext.php

```diff
@@ -257,6 +261,94 @@ public function updateProductOptions(string $productReference, TableNode $table)
         }
     }
 
+    /**
+     * @When I update product :productReference shipping information with following values:
+     *
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function updateProductShipping(string $productReference, TableNode $table): void
+    {
+        $data = $table->getRowsHash();
+        $productId = $this->getSharedStorage()->get($productReference);
+
+        try {
+            $command = new UpdateProductShippingCommand($productId);
+            $unhandledData = $this->setUpdateShippingCommandData($data, $command);
+
+            Assert::assertEmpty(
+                $unhandledData,
+                sprintf('Not all provided values handled in scenario. %s', var_export($unhandledData))
+            );
+
+            $this->getCommandBus()->handle($command);
+        } catch (ProductException $e) {
+            $this->lastException = $e;
+        }
+    }
+
+    /**
+     * @param array $data
+     * @param UpdateProductShippingCommand $command
+     *
+     * @return array values that was provided, but wasn't handled
+     */
+    private function setUpdateShippingCommandData(array $data, UpdateProductShippingCommand $command): array
+    {
+        $unhandledValues = $data;
+
+        if (isset($data['width'])) {
+            $command->setWidth($data['width']);
+            unset($unhandledValues['width']);
```

Comment:
Oohh, that's a really nice improvement We should add this kind of checks in other tests 👍